stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -1,8 +1,8 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 # preserve HermOrSym wrapper
-eigencopy_oftype(A::Hermitian, S) = Hermitian(copy_similar(A, S), sym_uplo(A.uplo))
-eigencopy_oftype(A::Symmetric, S) = Symmetric(copy_similar(A, S), sym_uplo(A.uplo))
+eigencopy_oftype(A::Hermitian, S) = Hermitian(copy_similar(parent(A), S), sym_uplo(A.uplo))
+eigencopy_oftype(A::Symmetric, S) = Symmetric(copy_similar(parent(A), S), sym_uplo(A.uplo))
```

Comment:
You mean like this?

```suggestion
eigencopy_oftype(A::Hermitian, S) = Hermitian(copytrito!(similar(parent(A), S, size(A)), A, A.uplo), sym_uplo(A.uplo))
eigencopy_oftype(A::Symmetric, S) = Symmetric(copytrito!(similar(parent(A), S, size(A)), A, A.uplo), sym_uplo(A.uplo))
```

That indeed reduces the runtime by a factor of 2, as expected, and the compilation time by one third.

Reply:
I've implemented this now

Reply:
The reason I found a performance gap is what you found: it should call `copytrito!` on `A.data`, not on `A`!